### PR TITLE
refactor(map_loader)!: prefix package and namespace with autoware

### PR DESCRIPTION
**Pull request by a-maumau**
## Description
This PR will add the autoware prefix to `map_loader` package.

## Related links
- autoware-documentation: https://github.com/autowarefoundation/autoware-documentation/pull/617
- autoware-lanelet2-extension: https://github.com/autowarefoundation/autoware_lanelet2_extension/pull/31
- autoware_tools: https://github.com/autowarefoundation/autoware_tools/pull/122

**Parent Issue:**

- https://github.com/autowarefoundation/autoware/issues/4569

## How was this PR tested?
Checked with running the autoware sample-map.

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

**Comment by a-maumau** [comment]
This branch is forked from and contains the changes in https://github.com/autowarefoundation/autoware.universe/pull/8878#event-14334278932 .

**Comment by SakodaShintaro** [comment]
@a-maumau 
There are some conflicts. Could you revolve them?

**Comment by a-maumau** [comment]
@SakodaShintaro I have resolved the conflicts.

**Review by SakodaShintaro — APPROVED** [review]
I have confirmed `logging_simulator` works well.

- **Directory Name**
  - [x] Add `autoware_` as a prefix to the directory name.

- **packages.xml**
  - [x] Add `autoware_` to the `name` element.

- **CMakeLists.txt**
  - [x] Change the `project` name to `autoware_***`.
  - [x] Add `autoware::` to `PLUGIN`.

- **Header Files (.hpp)**
  - [x] Add `autoware::` to `namespace`.
  - [x] Add `autoware` folder as necessary and update include statements accordingly.

- **Source Files (.cpp)**
  - [x] Add `autoware::` inside `RCLCPP_COMPONENTS_REGISTER_NODE`.

- **Launch Files**
  - [x] Add `autoware_` before `find-pkg-share`.
  - [x] Change `node pkg="<pkgname>"` to `autoware_<pkgname>`.

- **Verification Points**
  - [x] Search `find-pkg-share` in autoware
  - [x] Check README file (especially json schema)
  - [x] Pay attention to complex cases like `sensor_launch` as seen in `gnss_poser`

LGTM

File: planning/autoware_static_centerline_generator/package.xml [hunk 1]
```diff
@@ -33,7 +34,7 @@
   <depend>autoware_vehicle_info_utils</depend>
   <depend>geometry_msgs</depend>
   <depend>global_parameter_loader</depend>
-  <depend>map_loader</depend>
+  <depend>osqp_interface</depend>
```
Comment (SakodaShintaro) [inline planning/autoware_static_centerline_generator/package.xml:37]: The following error occurred.

```bash
+ rosdep install -y --from-paths src --ignore-src --rosdistro humble
ERROR: the following packages/stacks could not have their rosdep keys resolved
to system dependencies:
autoware_static_centerline_generator: Cannot locate rosdep definition for [osqp_interface]
```

I believe that this line is not necessary.

**Review by SakodaShintaro — CHANGES_REQUESTED** [review]

File: planning/autoware_static_centerline_generator/package.xml (hunk 1 above)
Comment (a-maumau) [reply planning/autoware_static_centerline_generator/package.xml]: Thank you for pointing out!
I fixed it in https://github.com/autowarefoundation/autoware.universe/pull/8927/commits/6618d30cc70f8ceef1f7de0bdda9a00d40564b5a .

**Review by SakodaShintaro — APPROVED** [review]
- [x] `planning_simulator` with [sample_map](https://autowarefoundation.github.io/autoware-documentation/main/tutorials/ad-hoc-simulation/planning-simulation/)
- [x] `logging_simulator` with [sample_rosbag](https://autowarefoundation.github.io/autoware-documentation/main/tutorials/ad-hoc-simulation/rosbag-replay-simulation/)
- [x] `logging_simulator` with [awsim_gt_data(TIER IV INTERNAL)](https://drive.google.com/file/d/1BtH3Ry5lu-h85GHM-sYq_jNJLw-s0re6/view?usp=drive_link)

LGTM

**Review by takayuki5168 — APPROVED** [review]
LGTM for planning

**Review by satoshi-ota — APPROVED** [review]

**Pull request merged**